source/common/http/http1/codec_impl.cc

```diff
@@ -664,8 +688,13 @@ void ServerConnectionImpl::onBelowLowWatermark() {
   }
 }
 
-ClientConnectionImpl::ClientConnectionImpl(Network::Connection& connection, ConnectionCallbacks&)
-    : ConnectionImpl(connection, HTTP_RESPONSE, MAX_RESPONSE_HEADERS_KB) {}
+ClientConnectionImpl::ClientConnectionImpl(Network::Connection& connection,
```

Comment:
Any reason to not just remove the old constructor and default validate_header_values to false?

Though now that I think about it, I think we'd be better off only having the new constructor and not having a default value.  If we force all callers to pick a boolean we won't miss any code sites using the new code path.

It will probably mean tweaking a bunch of unit tests (you can set them all true) but hopefully you can search and replace there

Reply:
The main reason I hadn't done that was the pervasiveness of the change: it involves a _bunch_ of find/replace that really blew up the diff for the PR.  (Due to the fact that it involves changing `ServerConnectionImpl`, `ClientConnectionImpl`, `CodecClient`/`CodecClientProd`, `ConnectionManagerConfig`, and associated mocks, unit tests, and call sites.

If you think that much code churn is warranted though, I can certainly do that, and make all the callers default to `false` (and default the unit test callers to `true`). I know originally we had talked about an overloaded constructor like the one included here, which didn't lead to nearly as much churn.

Reply:
Yeah, that's actually a fair point about code churn given we'd want to go back to the single argument once we flip this to true and deprecate.

Do you feel confident from grepping around that you have all the call sites under source/...?  If not, how about a hacked local version with a RELEASE_ASSERT in the old constructors to make sure we didn't miss any, and we'll call it a day?

Reply:
Yep, I added `RELEASE_ASSERT(false, "");` to the `{Client,Server}ConnectionImpl` constructors and ran Envoy to perform the steps outlined in the PR description (in the "testing" section).  Worked great.